src/dtx/dtx_resync.c

```diff
@@ -656,6 +651,20 @@ dtx_resync(daos_handle_t po_hdl, uuid_t po_uuid, uuid_t co_uuid, uint32_t ver,
 	D_INIT_LIST_HEAD(&dra.tables.drh_list);
 	dra.tables.drh_count = 0;
 
+	if (dra.discard_version != dra.resync_version) {
```

Comment:
hmm, if this is intend to only start for non-discard case, this check does not seem right here.

let's say tgt1,2 on rank 1 get reintegrated.

tgt1 co_in_ver = 2.
tgt2 co_in_ver = 3
pool_map version is 3.

so resync ULT triggered from pool map update will be with resync_version = 3. 

then on tgt1, discard_version = 2,  resync_version = 3.  discard_version != resync_version, but this is for discard.

or I miss sth here?

Reply:
The comment may be not clear enough. "dra.discard_version == dra.resync_version" means it is for pure reintegration; otherwise, it means there are some other rebuild process (and may be together with reintegration). For former case, we do not need DTX re-index; otherwise, DTX re-index needs to be triggered.